src/ig/IGExporter.ts

```diff
@@ -1479,6 +1589,69 @@ export class IGExporter {
       }
     }
   }
+
+  // Add an extension for the R5 property IG.definition.page.source[x] if provided in configuration
+  addNameUrlAndPageSourceExtensionForR4(
+    page: ImplementationGuideDefinitionPage,
+    configPages: ImplementationGuideDefinitionPage[]
+  ): void {
+    const sourceExtensionUrl =
+      'http://hl7.org/fhir/5.0/StructureDefinition/extension-ImplementationGuide.definition.page.source';
+    const nameExtensionUrl =
+      'http://hl7.org/fhir/5.0/StructureDefinition/extension-ImplementationGuide.definition.page.name';
+    const configPage = configPages?.find(
+      p =>
+        (p.nameUrl || p.name).substring(0, (p.nameUrl || p.name).lastIndexOf('.')) ===
+        page.nameUrl.substring(0, page.nameUrl.lastIndexOf('.'))
+    );
+    if (configPage) {
+      if (configPage.sourceUrl) {
+        // If nameUrl and sourceUrl do not match, add the sourceUrl value to an extension
+        if (configPage.nameUrl !== configPage.sourceUrl) {
+          page.extension = (page.extension ?? []).concat({
+            url: sourceExtensionUrl,
+            valueUrl: configPage.sourceUrl
+          });
+        }
+
+        if (configPage.name) {
```

Comment:
Don't we only need to do this if `configPage.name && configPage.nameUrl !== configPage.name`?

Reply:
Honestly, I forget everything we decided that day in the lab, so I could be wrong here...

Reply:
I do think we decided that, but I don't actually think that can ever happen in reality. `configPage` is constructed in `addConfiguredPage` in the IGExporter as we convert the configuration to the IG. However, that function only ever adds `nameUrl`.

I _thought_ that if a `name` is set in the configuration, it was added in the `updateForR5` function when we add the R5 properties. However, it doesn't look like that is actually the case. That should probably be added (my bad that it isn't in there now). Once we do, I think this function can stay the same.

Reply:
OK.  I see that `addConfiguredPage` is only looking for `page.nameUrl`.  So really, it needs to work for `page.nameUrl` _or_ `page.name`, right?

I still think that if it's R4, but the person put a `name`, we only want to add the extension if the `name` is _different_ than the `nameUrl`.  And I do think that's possible if the author did something stupid like putting `nameUrl` _and_`name` in the config (with different values). But if they're the same, I don't think the `name` extension adds any value to the R4 representation.

Reply:
I think I partially over-complicated this because that check for if `configPage.name` is defined is used elsewhere in this function. And assuming the `name` is properly imported from configuration, the logic should be in place here.

I updated this section in [809e207](https://github.com/FHIR/sushi/pull/1189/commits/809e207e1a5963fe1bbbb9acc9fe0af202002595) by adding in the logic you suggested and I also tried to simplify the if/else cases. Let me know what you think.

I do still think the import of the configuration and the R5 export needs to be updated to handle if `name` is included in the configuration, but I have a couple questions about it, so maybe we can discuss that on Monday before I push it here.

Reply:
@cmoesel  as discussed earlier, I also pushed [382309b](https://github.com/FHIR/sushi/pull/1189/commits/382309b430d759663b037dd1b75dab5d0a6dc6a7), which properly imports `page.name` from a configuration file and sets the value in an R5 IG. This should not effect the handling for R4 since the logic updated in [809e207](https://github.com/FHIR/sushi/pull/1189/commits/809e207e1a5963fe1bbbb9acc9fe0af202002595) should correctly handle the cases for when `name` is present or not. This latest commit ([382309b](https://github.com/FHIR/sushi/pull/1189/commits/382309b430d759663b037dd1b75dab5d0a6dc6a7)) technically doesn't impact the functionality of this PR, but it was a bug in the R5 IG configuration export, so it is good that it's fixed now.